### PR TITLE
update LLVM version to 16 in bpf-linker install instructions

### PR DESCRIPTION
**Pull request by dmitris**

### CODE REVIEW

File: docs/book/start/development.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ cargo install bpf-linker
 ```
 
 If you are running **macos, or linux on any other architecture**, you need to
-install LLVM 15 first, then install the linker with:
+install LLVM 16 first (for example, with `brew install llvm`), then install the linker with:
```
Comment (tamird) [inline docs/book/start/development.md:23]: please reflow this, it looks to be the longest line in the file now.

Comment (dmitris) [reply docs/book/start/development.md]: done in [3615722](https://github.com/aya-rs/book/pull/126/commits/36157229f2f10027d519f8458c46ed7f0afb8d46)

**Comment by dmitris** [comment]
maybe instead of the "outdate-prone" `install LLVM 16`, we should say `install the current version of LLVM`? We can also add: `(you can consult https://github.com/aya-rs/bpf-linker/blob/main/.github/workflows/build-test.yml#L25 for the LLVM version used by bpf-linker)`.

**Comment by tamird** [comment]
> maybe instead of the "outdate-prone" `install LLVM 16`, we should say `install the current version of LLVM`? We can also add: "(you can consult https://github.com/aya-rs/bpf-linker/blob/main/.github/workflows/build-test.yml#L25 for the LLVM version used by bpf-linker)".

Yes, that would be better I think. That can also rot as that file changes, but more slowly.

**Comment by tamird** [comment]
On the other hand, we will soon require a custom LLVM because 16 doesn't contain various patches we'll soon need, and we'll move to a binary distribution at that time.

Let me know when you're happy with this and I'll merge it.

**Comment by dmitris** [comment]
> Let me know when you're happy with this and I'll merge it.

I think we can merge with "LLVM 16" if we change to the custom LLVM.  I understand it may be necessary but it's a bit sad - it was very convenient to be able to do the cross-compilation just with the brew'ed one 😄

**Pull request merged**